### PR TITLE
[5.4] Fix route bindings

### PR DESCRIPTION
**Pull request by themsaid**
This is another approach to fix https://github.com/laravel/framework/pull/16926

### CODE REVIEW

**Comment by cgoosey1** [comment]
This seems a lot more elegant than my solution in #16926, can't see any issues address in mine that haven't been addressed in this one (though haven't thought through too much the $values[$key - $instancesCount] maintaining the order of non object dependencies.)

File: src/Illuminate/Routing/RouteDependencyResolverTrait.php [hunk 1]
```diff
@@ -57,10 +66,9 @@ public function resolveMethodDependencies(array $parameters, ReflectionFunctionA
      *
      * @param  \ReflectionParameter  $parameter
      * @param  array  $parameters
-     * @param  array  $originalParameters
      * @return mixed
      */
-    protected function transformDependency(ReflectionParameter $parameter, $parameters, $originalParameters)
+    protected function transformDependency(ReflectionParameter $parameter, $parameters)
```
Comment (cgoosey1) [inline src/Illuminate/Routing/RouteDependencyResolverTrait.php]: same feedback as appeared in mine, this can be type hinted :P

Comment (themsaid) [inline src/Illuminate/Routing/RouteDependencyResolverTrait.php]: ¯\_(ツ)_/¯ :D

Comment (cgoosey1) [inline src/Illuminate/Routing/RouteDependencyResolverTrait.php]: `protected function transformDependency(ReflectionParameter $parameter, array $parameters)`

**Comment by vlakoff** [comment]
How about applying this bug fix to the previous, supposedly still supported versions?

**Comment by GrahamCampbell** [comment]
It's probably too controversial for earlier versions IMO.

**Pull request merged**